tests/api/test_cluster_descriptor.cpp

```diff
@@ -0,0 +1,140 @@
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+#include "tests/test_utils/generate_cluster_desc.hpp"
+
+#include "device/pcie/pci_device.hpp"
+#include "device/tt_cluster_descriptor.h"
+
+
+std::unique_ptr<tt_ClusterDescriptor> get_cluster_desc() {
+
+    std::vector<int> pci_device_ids = PCIDevice::enumerate_devices();
+    std::set<int> pci_device_ids_set (pci_device_ids.begin(), pci_device_ids.end());
+
+    // TODO: This test requires knowledge of the device architecture, which should not be true.
+    tt::ARCH device_arch = tt::ARCH::GRAYSKULL;
+    if (!pci_device_ids.empty()) {
+        int physical_device_id = pci_device_ids[0];
+        PCIDevice pci_device (physical_device_id, 0);
+        device_arch = pci_device.get_arch();
+    }
+
+    // TODO: Make this test work on a host system without any tt devices.
+    if (pci_device_ids.empty()) {
+        std::cout << "No Tenstorrent devices found. Skipping test." << std::endl;
+        return nullptr;
+    }
+
+    // TODO: remove getting manually cluster descriptor from yaml.
+    std::string yaml_path = test_utils::GetClusterDescYAML();
+    // TODO: Remove the need to do this, allow default constructor to construct with all chips.
+    std::unique_ptr<tt_ClusterDescriptor> cluster_desc;
+    if (device_arch == tt::ARCH::GRAYSKULL) {
+        cluster_desc = tt_ClusterDescriptor::create_for_grayskull_cluster(pci_device_ids_set, pci_device_ids);
+    } else {
+        cluster_desc = tt_ClusterDescriptor::create_from_yaml(yaml_path);
+    }
+
+    return cluster_desc;
+}
+
+TEST(ApiClusterDescriptorTest, BasicFunctionality) {
+
+    std::unique_ptr<tt_ClusterDescriptor> cluster_desc = get_cluster_desc();
+
+    if (cluster_desc == nullptr) {
+        return;
+    }
+
+    std::unordered_set<chip_id_t> all_chips = cluster_desc->get_all_chips();
+    std::unordered_map<chip_id_t, std::uint32_t> harvesting_for_chips = cluster_desc->get_harvesting_info();
+    std::unordered_map<chip_id_t, eth_coord_t> eth_chip_coords = cluster_desc->get_chip_locations();
+    std::unordered_map<chip_id_t, chip_id_t> local_chips_to_pci_device_id = cluster_desc->get_chips_with_mmio();
+    std::unordered_set<chip_id_t> local_chips;
+    for (auto [chip, _]: local_chips_to_pci_device_id) {
+        local_chips.insert(chip);
+    }
+    std::unordered_set<chip_id_t> remote_chips;
+    for (auto chip : all_chips) {
+        if (local_chips.find(chip) == local_chips.end()) {
+            remote_chips.insert(chip);
+        }
+    }
+}
```

Comment:
Does this test need any kind of check, or is it just intended to exercise the code?

Reply:
Well it checks that the code doesn't throw :)
My intention for these api tests is for them to be more focused on offering usage examples and documenting API rather than testing functionality.